src/pymor/core/cache.py

```diff
@@ -186,7 +186,14 @@ def set(self, key, value):
         if has_key:
             getLogger('pymor.core.cache.DiskRegion').warning('Key already present in cache region, ignoring.')
             return
-        self._cache.set(key, value)
+        try:
+            self._cache.set(key, value)
+        except NotImplementedError:
```

Comment:
Sorry, I thought the `NotImplementedError` would be converted by `dumpy` into something meaningful like a `PicklingError`. Catching a `NotImplementedErrror`, which also might be raised somewhere else inside `set`, is probably not good practice. 

There does not seem to be an official way to mark an object as unpicklable. So I would propose the following:

- Add `UnpicklableObject` to `pymor.core.exceptions`
- Use `raise UnpicklableObject(self)` in `__getstate__`
- Catch `UnpicklableObject` and unconditionally produce a warning reporting which object raised the exception.